kernel/interrupts/src/lib.rs

```diff
@@ -9,3 +9,10 @@
 mod arch;
 
 pub use arch::*;
+
+#[derive(Debug, PartialEq, Eq)]
+#[repr(C)]
+pub enum EoiBehaviour {
+    CallerMustSignalEoi,
+    HandlerHasSignaledEoi,
```

Comment:
add enum-level docs and docs for each variant.

Reply:
I think we could also improve the names of the variants, specifically the first one. It's a bit counter-intuitive to think of an interrupt handler as having a "caller", since they are typically described as entry points that the hw jumps to directly. I understand we are now wrapping it, but it's still not "called" in the traditional sense of a function.

Perhaps it should be something like:
```rust
pub enum EoiBehaviour {
    HandlerSentEoi,
    HandlerDidNotSendEoi,
}
```
which has no ambiguity.

Reply:
great!